lib/utils/hash.hpp

```diff
@@ -0,0 +1,12 @@
+namespace hyped::utils {
+
+inline void hash_combine(std::size_t& seed) { }
+
+template <class T>
+inline void hash_combine(std::size_t & seed, const T& v)
+{
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+};
```

Comment:
I don't like this snippet because it looks so arbitrary; can we not just depend on boost? Just require it to be present on the host like we do with eigen and we should be good. @kshxtij